test/cholesky/cholesky.cpp

```diff
@@ -85,12 +83,12 @@ void entrypoint(void *arg) {
         // Taking this malloc out from triSolve method
         for(int j = k + 1 ; j < numTiles ; ++j ) {
             TileBlock *currPivotColumnTile = lkji[j][k][k+1];
-            currPivotColumnTile->matrixBlock = (double**) malloc (sizeof(double*)*tileSize);
+            currPivotColumnTile->matrixBlock = new double*[tileSize];
             for(i = 0; i < tileSize; ++i)
-                currPivotColumnTile->matrixBlock[i] = (double*) malloc (sizeof(double)*tileSize);
+                currPivotColumnTile->matrixBlock[i] = new double[tileSize];
         }
 
-        finish ([=]() {
+        HCLIB_FINISH {
```

Comment:
Yes, it avoid creating an additional lambda/closure. Besides the efficiency issue, it's also nice from a maintenance perspective because you don't have to explicitly list captures or worry about unexpected captures due to a catch-all.